contribute/style-guides/testing.md

```diff
@@ -0,0 +1,251 @@
+  return (
+    <Select
+      inputId={inputId}
+      value={value}
+      options={options}
+      onChange={(val) => onChange(val.value as OrgRole)}
+      placeholder="Choose role..."
+      aria-label={ariaLabel}
+      autoFocus={autoFocus}
+      {...restProps}
+    />
+  );
+}
+```
+
+### Querying the Select Component
+
+There are a few methods to query the `Select` component:
+
+1. One approach is to explicitly pass the `aria-label` prop and then use the `getByRole` method:
+
+```tsx
+describe('OrgRolePicker', () => {
+  it('should render the picker', () => {
+    setup(<OrgRolePicker value={OrgRole.Admin} aria-label={'Role picker'} onChange={() => {}} />);
+    expect(screen.getByRole('combobox', { name: 'Role picker' })).toBeInTheDocument();
+  });
+});
+```
+
+2. Alternatively, add a `label` element and provide the `htmlFor` prop. In this scenario, a matching `inputId` should be passed to the `Select` component:
+
+```tsx
+describe('OrgRolePicker', () => {
+  it('should render the picker', () => {
+    setup(
+      <>
+        <label htmlFor={'role-picker'}>Role picker</label>
+        <OrgRolePicker value={OrgRole.Admin} inputId={'role-picker'} onChange={() => {}} />
+      </>
+    );
+    expect(screen.getByRole('combobox', { name: 'Role picker' })).toBeInTheDocument();
+  });
+});
+```
+
+It's also possible to use `*ByLabelText` queries. However, the `*ByRole` queries are [more robust](https://testing-library.com/docs/queries/bylabeltext/#name) and are generally recommended over the former.
+
+### Testing the Display of Correct Options
+
+At times, it might be necessary to verify that the `Select` component is displaying the correct options. In such instances, the best solution is to click the `Select` component and match the desired option using the `*ByText` query.
+
+```tsx
+it('should have an "Editor" option', async () => {
+  const { user } = setup(<OrgRolePicker value={OrgRole.Admin} aria-label={'Role picker'} onChange={() => {}} />);
+  await user.click(screen.getByRole('combobox', { name: 'Role picker' }));
+  expect(screen.getByText('Editor')).toBeInTheDocument();
+});
+```
+
+### Selecting an option
+
+To simplify the process of selecting an option from a `Select` component, there is a `selectOptionInTest` utility function. This function is a wrapper over the [react-select-event](https://testing-library.com/docs/ecosystem-react-select-event/) package.
+
+```tsx
+it('should select an option', async () => {
+  const mockOnChange = jest.fn();
+  setup(<OrgRolePicker value={OrgRole.Admin} aria-label={'Role picker'} onChange={mockOnChange} />);
+  await selectOptionInTest(screen.getByRole('combobox', { name: 'Role picker' }), 'Viewer');
+  expect(mockOnChange).toHaveBeenCalledWith('Viewer');
+});
+```
+
+## Mocking Objects and Functions
+
+### Mocking the `window` Object and Its Methods
+
+There are several approaches to mock the `window` object. The most common methods are using `Object.defineProperty` and Jest spies.
```

Comment:
is there an advantage to one over the other? I would just keep the the jest spies approach.

Reply:
Using Jest spies has a built-in mechanism for restoring mocks, so probably a better option. I've included the `Object.defineProperty` approach as it's used a lot in our codebase and I don't see anything wrong with it, but probably having only one of them as recommended is a good idea.

Reply:
You do what you think's best, but I think it's a good idea for this doc to be more of an opinionated/"here's how to test" guide. I think jest spies are the ones worth pushing :)

Reply:
yeah, makes sense!